hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/exception/QuotaExceededException.java

```diff
@@ -13,31 +13,75 @@
 import org.eclipse.hawkbit.repository.model.BaseEntity;
 
 /**
- * Thrown if too many entries are added to repository.
+ * Thrown if quota is exceeded (too many entries, file size, storage size).
  *
  */
 public final class QuotaExceededException extends AbstractServerRtException {
 
     private static final long serialVersionUID = 1L;
 
+    private static final String KB = "KB";
+    private static final String MB = "MB";
+
     private static final String ASSIGNMENT_QUOTA_EXCEEDED_MESSAGE = "Quota exceeded: Cannot assign %s more %s entities to %s '%s'. The maximum is %s.";
+    private static final String MAX_ARTIFACT_SIZE_EXCEEDED = "Maximum artifact size (%s) exceeded.";
+    private static final String MAX_ARTIFACT_SIZE_TOTAL_EXCEEDED = "Storage quota exceeded, %s left.";
+
+    private final QuotaType quotaType;
+
+    private final long exceededQuotaValue;
+
+    /**
+     * Enum that describes the types of quota that may be exceeded
+     */
+    public enum QuotaType {
```

Comment:
I would better create three separate exception classes: `AssignmentQuotaExceededException`, `StorageQuotaExceededException` and `FileSizeQuotaExceededException` in order to remove conditional statements and unnecessary constructors/usage of member variables